### PR TITLE
fix(connector api): handle bad tls config file conversion errors

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-12581

Release version: v/e5.7.2

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_connector/src/emqx_connector_api.erl [hunk 1]
```diff
@@ -795,6 +795,8 @@ maybe_unwrap(RpcMulticallResult) ->
 redact(Term) ->
     emqx_utils:redact(Term).
 
+map_to_json({bad_ssl_config, Map}) when is_map(Map) ->
+    map_to_json(Map);
```
Comment (keynslug) [inline apps/emqx_connector/src/emqx_connector_api.erl:799]: I'd argue this is a weird place for that, given that `emqx_connector` already tries to fit [all other validation-ish errors](https://github.com/emqx/emqx/blob/c17c7884380313d1b2075da90fa9ec1cada9dc48/apps/emqx_connector/src/emqx_connector.erl#L541-L545) into a common format. So one would naturally expect this is taken care of there as well.

File: apps/emqx_connector/test/emqx_connector_api_SUITE.erl [hunk 2]
```diff
@@ -718,6 +718,36 @@ t_create_with_bad_name(Config) ->
...
+                uri(["connectors"]),
+                Conf,
+                Config
+            ),
+            ?assertMatch(
+                #{
+                    <<"file_read">> := <<"enoent">>,
+                    <<"pem_check">> := <<"invalid_pem">>,
+                    <<"which_options">> := [[<<"cacertfile">>]]
+                },
```
Comment (keynslug) [inline apps/emqx_connector/test/emqx_connector_api_SUITE.erl:742]: Nit. Might be out-of-scope but I definitely feel this needs to be made easier to understand, currently it sounds cryptic IMO.

File: apps/emqx_connector/src/emqx_connector_api.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_connector/src/emqx_connector_api.erl:799]: Makes sense.  Adapting the [originating error](https://github.com/emqx/emqx/blob/c17c7884380313d1b2075da90fa9ec1cada9dc48/apps/emqx_connector/src/emqx_connector.erl#L355) might be better, indeed.

File: apps/emqx_connector/src/emqx_connector_ssl.erl [hunk 3]
```diff
@@ -48,3 +48,25 @@ new_ssl_config(#{<<"ssl">> := _} = Config, NewSSL) ->
...
 new_ssl_config(Config, _NewSSL) ->
     Config.
+
+map_bad_ssl_error(#{pem_check := invalid_pem} = TLSLibError) ->
+    #{which_options := Paths, file_read := Reason} = TLSLibError,
+    #{
+        kind => validation_error,
+        reason => <<"bad_ssl_config">>,
+        bad_fields => Paths,
+        file_read_error => Reason
```
Comment (keynslug) [inline apps/emqx_connector/src/emqx_connector_ssl.erl:58]: Tiny nit. Maybe something like `details => emqx_utils:format("Failed to access certificate / key file: ~s", [emqx_utils:explain_posix(Reason)])` to make the error more consistent?

**Review by keynslug — DISMISSED** [review]

**Review by keynslug — DISMISSED** [review]

**Review by keynslug — APPROVED** [review]

**Comment by thalesmg** [comment]
Thanks for the review, @keynslug .  :beers:

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2737)

**Pull request merged**